Source/controls/plrctrls.cpp

```diff
@@ -717,6 +739,25 @@ int FindClosestInventorySlot(Point mousePos)
 	return bestSlot;
 }
 
+Point FindClosestStashSlot(Point mousePos)
+{
+	int shortestDistance = std::numeric_limits<int>::max();
+	Point bestSlot = {};
+	mousePos += Displacement { -INV_SLOT_HALF_SIZE_PX, INV_SLOT_HALF_SIZE_PX };
+
+	for (int y = 0; y < 10; y++) {
+		for (int x = 0; x < 10; x++) {
+			int distance = mousePos.ManhattanDistance(GetStashSlotCoord({ x, y }));
+			if (distance < shortestDistance) {
+				shortestDistance = distance;
+				bestSlot = { x, y };
+			}
+		}
+	}

```

Comment:
Loops like this which use `x,y` as a point can use `PointsInRectangleRange`
```suggestion
	for (auto point : PointsInRectangleRange{{0, 0}, {10, 10}}) {
		int distance = mousePos.ManhattanDistance(GetStashSlotCoord(point));
		if (distance < shortestDistance) {
			shortestDistance = distance;
			bestSlot = point;
		}
	}
```

could even potentially use `std::min_element` but given the derived attribute it doesn't make the logic clearer.

Reply:
There are probably lots of opotunities like this all around the codebase :thinking: